### PR TITLE
[WIP] Define providers services as lazy

### PR DESCRIPTION
**Pull request by akerouanton**

### CODE REVIEW

**Comment by NiR-** [comment]
This patch does not work :( Without proper definition of the service class, it's not possible to declare cache providers as lazy.

File: src/DummyAdapter.php [hunk 1]
```diff
@@ -21,7 +21,7 @@
  *
  * @author Tobias Nyholm <tobias.nyholm@gmail.com>
  */
-final class DummyAdapter implements CacheItemPoolInterface, LoggerAwareInterface
+class DummyAdapter implements CacheItemPoolInterface, LoggerAwareInterface
```
Comment (prisis) [inline src/DummyAdapter.php:24]: why did you remove the final?

Comment (akerouanton) [reply src/DummyAdapter.php:24]: It's a hack around lazy services whose class can't be final. But it doesn't work properly in dev env due to the proxy classes generated by the [ProxyFactory](https://github.com/php-cache/cache-bundle/blob/master/src/DataCollector/ProxyFactory.php).

**Comment by jewome62** [comment]
Hello !

I have a problem into SF4 & docker build
I have not env variable as Database_url or Redis_url into docker build

I have add 
`env(REDIS_URL): ''`

But in contrary of Doctrine, the Adapter connect into services into cache:warmup.
So that PR could fix my problem.

Do you need some help on this PR ?
What are the blocking points?
